components/formstack/formstack.app.js

```diff
@@ -0,0 +1,83 @@
+const axios = require("axios");
+
+module.exports = {
+  type: "app",
+  app: "formstack",
+  propDefinitions: {
+    formId: {
+      type: "string",
+      label: "Forms",
+      optional: true,
+      async options() {
+        const options = [];
+        const results = await this.getForms();
+        for (const form of results) {
+          options.push({ label: form.name, value: form.id });
+        }
+        return options;
+      },
+    },
+  },
+  methods: {
+    monthAgo() {
+      const monthAgo = new Date();
+      monthAgo.setMonth(monthAgo.getMonth() - 1);
+      return monthAgo;
+    },
+    _getBaseURL() {
+      return "https://www.formstack.com/api/v2"
+    },
+    _getAuthHeaders() {
+      return {
+        'Authorization': `Bearer ${this.$auth.oauth_access_token}`,
+        'Content-Type': 'application/json',
+      };
+    },
+    async createHook({ id, url }) {
+      const config = {
+        method: "POST",
+        url: `${this._getBaseURL()}/form/${id}/webhook.json`,
+        headers: this._getAuthHeaders(),
+        params: {
+          url,
+          content_type: "json",
+          handshake_key: this.$auth.oauth_refresh_token,
+        }
+      }
+      return (await axios(config)).data;
+    },
+    async deleteHook({ hookId }) {
+      const config = {
+        method: "DELETE",
+        url: `${this._getBaseURL()}/webhook/${hookId}.json`,
+        headers: this._getAuthHeaders(),
+      }
+      return await axios(config);
+    },
+    async getForms() {
```

Comment:
this might be cleaner as a wrapper around just the api call, so it takes the folders, page and per_page params.  Then you handle the paging logic in the component code.

Reply:
Since this function is used both for props and for components, moving the pagination from the function to both of those places would create some duplication. Other reviewers have sometimes recommended consolidating code to reduce cross-component duplication, but I suppose it's just a matter of preference. If your preference is to leave pagination out of app file methods, I'll extrapolate it to both places.